src/octree/types.rs

```diff
@@ -31,46 +31,59 @@ pub(in crate::octree) enum NodeChildrenArray<T: Default> {
 #[cfg_attr(feature = "serialization", derive(Serialize, Deserialize))]
 pub(in crate::octree) struct NodeChildren<T: Default> {
     /// The key value to signify "no child" at a given slot
-    pub(in crate::octree) default_key: T,
+    pub(in crate::octree) no_child_marker_key: T,
 
     /// The contained child key values
     pub(in crate::octree) content: NodeChildrenArray<T>,
 }
 
 pub trait VoxelData {
-    fn new(r: u8, g: u8, b: u8, a: u8, user_data: u32) -> Self;
+    fn new(color: Albedo, user_data: u32) -> Self;
     /// The color to display during raytracing 0-255 RGBA
-    fn albedo(&self) -> [u8; 4];
+    fn albedo(&self) -> Albedo;
     /// User defined data
     fn user_data(&self) -> u32;
-    /// determines if the voxel is to be hit by rays in the raytracing algorithms
+    /// Determines if the voxel is to be hit by rays in the raytracing algorithms
     fn is_empty(&self) -> bool {
-        [0, 0, 0, 0] == self.albedo() && 0 == self.user_data()
+        self.albedo().is_transparent() && self.user_data() == 0
     }
     /// Implementation to clear the contained data, as well as albedo
     fn clear(&mut self);
 }
 
-impl VoxelData for u32 {
-    fn new(r: u8, g: u8, b: u8, a: u8, _user_data: u32) -> Self {
-        r as u32 & 0x000000FF
-            | ((g as u32 & 0x000000FF) << 8)
-            | ((b as u32 & 0x000000FF) << 16)
-            | ((a as u32 & 0x000000FF) << 24)
+impl VoxelData for Albedo {
+    fn new(color: Albedo, _user_data: u32) -> Self {
+        color
     }
-    fn albedo(&self) -> [u8; 4] {
-        [
-            (self & 0x000000FF) as u8,
-            ((self & 0x0000FF00) >> 8) as u8,
-            ((self & 0x00FF0000) >> 16) as u8,
-            ((self & 0xFF000000) >> 24) as u8,
-        ]
+
+    fn albedo(&self) -> Albedo {
+        self.clone()
     }
+
     fn user_data(&self) -> u32 {
-        0
+        0u32.into()
     }
+
     fn clear(&mut self) {
-        *self = 0;
+        self.r = 0;
+        self.r = 0;
+        self.b = 0;
+        self.a = 0;
+    }
+}
+
+impl From<u32> for Albedo {
+    fn from(value: u32) -> Self {
+        let a = (value & 0x000000FF) as u8;
+        let b = ((value & 0x0000FF00) >> 8) as u8;
+        let g = ((value & 0x00FF0000) >> 16) as u8;
+        let r = ((value & 0xFF000000) >> 24) as u8;
```

Comment:
Just a sidenote: This implementation is actually changing the Byte order which is currently implemented. That is fine though because I would prefer the new order, but the tests are using opposite byte orders. 

See: https://github.com/davids91/shocovox/pull/25/files#r1622491246

Reply:
I thought so... that was the reason why I had to add the last byte to e.g. "FF0000" for the "alpha"/"opacity"... I figured it out after I changed the tests, when I was lying in bed, lol